### PR TITLE
list.py: get by date range method, app.py: separate get everything an…

### PR DESCRIPTION
**Pull request by kamigerami**
added get_by_date_range

### CODE REVIEW

File: timereport/app.py [hunk 1]
```diff
@@ -54,10 +54,25 @@ def index():
...
-
         # python-backend
-        response = get_user_by_id(f'{python_backend_url}/user', payload.get('user_id'))
-        # post back to slack as a code formatted output ``` data ```
-        for r in response:
+
+        # get everything
+        get_by_user = get_user_by_id(f'{python_backend_url}/user', payload.get('user_id'))
+        for r in get_by_user:
             slack_responder(response_url, f'```{str(r)}```')
```
Comment (parberge) [inline timereport/app.py]: What does this do?

Comment (kamigerami) [reply timereport/app.py]: it will basically just post to slack response_url with the clasic ``` stuff here ```formatting

**Comment by kamigerami** [comment]
https://github.com/codelabsab/timereport/issues/13
https://github.com/codelabsab/timereport/issues/10

**Comment by kamigerami** [comment]
https://github.com/codelabsab/timereport/issues/17

File: timereport/app.py [hunk 2]
```diff
@@ -34,30 +33,86 @@ def index():
...
-    command = params.pop(0)
-    auth_token = payload['team_id']
-
+    channel_id = payload.get('channel_id')
+    user_id = payload.get('user_id')
+    token = payload.get('token')
     # needs to do something on True or False return statement
-    verify_token(payload['token'])
+
+    # this payload token we receive is not the token we use to communicate with slack
```
[the remark this answers is not in the document]
Comment (kamigerami) [reply timereport/app.py]: well I figured I should have an explanation for removing that code .. or commenting that out rather...

File: timereport/app.py [hunk 3]
```diff
@@ -34,30 +33,86 @@ def index():
...
+        # get everything
+        get_by_user = get_user_by_id(f'{python_backend_url}/user', payload.get('user_id'))
+        if isinstance(get_by_user, tuple):
+            app.log.debug(f'Failed to return anything: {get_by_user[1]}')
+        else:
+            for r in get_by_user:
+                slack_responder(response_url, f'```{str(r)}```')
+
+        # get by date
+        event_date = ''.join(params[-1:])
```
[the remark this answers is not in the document]
Comment (kamigerami) [reply timereport/app.py]: yeah difficult to know what type of data that gets in

File: timereport/chalicelib/lib/list.py [hunk 4]
```diff
@@ -18,6 +18,7 @@ def get_user_by_id(url, user_id):
     log.debug(f'python backend')
     log.debug(f'Response code is: {res.status_code}')
     log.debug(f'Url is: {res.url}')
+    log.debug(f'function is get_user_by_id')
```
[the remark this answers is not in the document]
Comment (kamigerami) [reply timereport/chalicelib/lib/list.py]: sure .. just so we know if we are inside that particular function or not

File: timereport/chalicelib/lib/slack.py [hunk 5]
```diff
@@ -1,6 +1,26 @@
 import os
 import botocore.vendored.requests.api as requests
 from urllib.parse import parse_qs
+import logging
+
+log = logging.getLogger(__name__)
+
+def slack_client_responder(token, channel_id, user_id, attachment):
+    log.debug('we are now inside slack_client_responder')
```
[the remark this answers is not in the document]
Comment (kamigerami) [reply timereport/chalicelib/lib/slack.py]: well it is if you are trying to debug the code and want to know what got executed at what time

File: timereport/chalicelib/lib/slack.py [hunk 6]
```diff
@@ -1,6 +1,26 @@
...
+
+    headers = {'Content-Type': 'application/json; charset=utf-8', 'Authorization': f'Bearer {token}'}
+    res = requests.post(url='https://slack.com/api/chat.postMessage',
+                        json={'channel': channel_id, 'text': f'{user_id} from slack.py', 'attachments': attachment},
+                        headers=headers
+                        )
+    log.debug(f'Response code is: {res.status_code}')
+    log.debug(f'Url is: {res.url}')
+    log.debug(f'function is slack_client_responder')
```
[the remark this answers is not in the document]
Comment (kamigerami) [reply timereport/chalicelib/lib/slack.py]: same thing here .. these won't bother anyone unless you turn debug on .. and we can remove them when we are done with development ...  It's very cumbersome to add these..debug while coding and then remove and push..then add back again etc ..everytime... 

can we agree on keeping them for now during development and then clean everything up when everybody is up to speed on the codebase etc ?

File: timereport/app.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (parberge) [reply timereport/app.py]: I think the correct forum to write those are in the PR message

File: timereport/chalicelib/lib/list.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (parberge) [reply timereport/chalicelib/lib/list.py]: Why is that necessary? You could use logging formatting to include what file/package the log message came from

File: timereport/chalicelib/lib/slack.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (parberge) [reply timereport/chalicelib/lib/slack.py]: I think there are other debugging techniques to see if functions get executed or not. And honestly if we can't know if a function or not is used we have a problem 😁

File: timereport/chalicelib/lib/slack.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (parberge) [reply timereport/chalicelib/lib/slack.py]: The problem is that to me the code gets kind of harder to read, not easier. And you are usually reading the code more often than executing. But I don't have a strong opinion. If you want to keep these go ahead

File: timereport/chalicelib/lib/list.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kamigerami) [reply timereport/chalicelib/lib/list.py]: interesting... please provide example...

[the remark this answers is not in the document]
Comment (parberge) [reply timereport/chalicelib/lib/list.py]: If you read this then you will understand:
https://docs.python.org/3/howto/logging-cookbook.html#using-logging-in-multiple-modules

File: tests/test_timereport.py [hunk 7]
```diff
@@ -19,7 +19,7 @@ def test_parsing_config():
 
 
 def test_slack_payload_extractor():
-    fake_data = slack_payload_extractor({'body': 'foo=bar&text=fake+text'})
+    fake_data = slack_payload_extractor('foo=bar&text=fake+text')
```
Comment (parberge) [inline tests/test_timereport.py]: Why is the key `body` removed?

**Comment by kamigerami** [comment]
do we merge this?

**Comment by parberge** [comment]
Can we tag the current master with a tag (with version 1.0.0 for example)
and then merge this to master instead?

**Pull request merged**